### PR TITLE
docs: move internal auth SDK examples SDKCF-4169

### PR DESCRIPTION
**Pull request by apparition47**
# Description

Move internal SDK usage references/examples for `registerPreference` to our internal docs.

## Links

mitigates SDKCF-4061

# Checklist
- [ ] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [ ] I wrote/updated tests for new/changed code
- [ ] I removed all sensitive data **before every commit**, including API endpoints and keys
- [ ] I ran `fastlane ci` without errors

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -37,6 +37,8 @@ To use the module you must set the following values in your app's `Info.plist`:
 | `InAppMessagingAppSubscriptionID` | your_subscription_key |
 | `InAppMessagingConfigurationURL` | Endpoint for fetching the configuration |
 
+⚠️ Targeted campaigns not showing? When using the staging dashboard and internal RAuthentication/Identity SDK's staging environment, you must specify the corresponding staging values here.
```
Comment (imaginaris) [inline README.md]: The question is - is it better to have the info here (near setup info) or in Troubleshooting section?

Comment (donnie-jp) [reply README.md]: It could be something like
> ⚠️ Targeted campaigns not showing? See our [troubleshooting guide](#troubleshooting-campaigns).

Comment (donnie-jp) [reply README.md]: But actually, we don't expose staging dashboard/endpoints to customers do we? I thought for "staging" they just had to set up a different dashboard app group - is that what's meant here? It isn't clear.

**Review by donnie-jp — CHANGES_REQUESTED** [review]
We don't expose staging dashboard/endpoints to customers do we? I thought for "staging" they just had to set up a different dashboard app group - is that what's meant here? It isn't clear.

File: README.md (hunk 1 above)
Comment (apparition47) [reply README.md]: Going to revert this line for now. Wasn't aware that our staging was internal

**Review by donnie-jp — APPROVED** [review]
👍

File: README.md [hunk 2]
```diff
@@ -111,38 +111,16 @@ RInAppMessaging.logEvent(CustomEvent(withName: "any_event_name_here", withCustom
...
-#### Using the internal Identity SDK
-
-```swift
-let preference = IAMPreferenceBuilder()
-    .setIDTrackingIdentifier("998236")
-    .build()
-
-RInAppMessaging.registerPreference(preference)
-```
+Preferences are not persisted so this function needs to be called on every launch.
```
Comment (imaginaris) [inline README.md:123]: 👍

**Review by imaginaris — APPROVED** [review]

**Pull request merged**